### PR TITLE
fix: flaky test in TestProxySocketsAdminFacade

### PR DESCRIPTION
**Pull request by kian99**
## Description

This PR fixes a flaky test case in `TestProxySocketsAdminFacade`.

The test was incorrectly cancelling the context. When the proxySockets function encounters an error like a failure to connect to a Juju controller it returns with an error. Otherwise it returns a nil error when the context is cancelled. Here we were expecting the former for one of the tests cases but sometimes encountering the latter in a race condition.

The test failed often when running `go test -timeout 30s -run ^TestProxySocketsAdminFacade$ github.com/canonical/jimm/v3/internal/rpc -count=1000`, after the fix I didn't encounter any failures even after increasing count to 10 000.

## Engineering checklist
*Check only items that apply*

- [ ] Documentation updated
- [x] Covered by unit tests
- [ ] Covered by integration tests

### CODE REVIEW

**Review by SimoneDutto — APPROVED** [review]
beating the flaky tests allegation

File: internal/rpc/apiproxy_test.go [hunk 1]
```diff
@@ -275,8 +277,8 @@ func TestProxySocketsAdminFacade(t *testing.T) {
 			go func() {
 				defer wg.Done()
 				err = rpc.ProxySockets(ctx, helpers)
-				if test.expectConnectTofail {
-					c.Assert(err, qt.ErrorMatches, "failed to connect to controller: controller connection error")
+				if test.expectProxyError {
```
Comment (alesstimec) [inline internal/rpc/apiproxy_test.go]: could we simplify this and do 
```
if test.proxyErrorMessage != ""
```
and lose the bool?

**Review by alesstimec — APPROVED** [review]
LGTM with a question/suggestion

File: internal/rpc/apiproxy_test.go (hunk 1 above)
Comment (kian99) [reply internal/rpc/apiproxy_test.go]: Yes, done.

**Review by ale8k — APPROVED** [review]

**Pull request merged**